openspoor/visualisations/trackmap.py

```diff
@@ -17,60 +19,68 @@ class PlotObject(ABC):
     A parent class for every object that can be plotted on a TrackMap
     """
 
+    def __init__(self, data, popup):
+        if popup is None or isinstance(popup, list):
+            self.popup = popup
+        elif isinstance(popup, str):
+            self.popup = [popup]
+        else:
+            TypeError('Unknown type for popup in linestrings')
+
+        if isinstance(data, gpd.GeoDataFrame):
+            self.data = data
+        elif isinstance(data, str):
+            self.data = gpd.read_file(data)
+        else:
+            raise TypeError('Provide either a geopandas dataframe or a file location of a csv to show')
+        self.data = self.data.to_crs('EPSG:4326')
+        if self.popup:
+            self.data = self.data.set_index(self.popup)
+
     @abstractmethod
     def add_to(self, m) -> None:
         """
         A base function that should be overwritten with logic for every plottable object.
+
         :param m: A TrackMap object, to which this element should be added.
         :return: None
         """
         raise NotImplementedError('This needs to be set in implementation classes')
 
-    def _convert_pandas_to_geopandas(self, df):
-        # If this is a pandas dataframe with Rijksdriehoek in it, convert it to GPS
-        first_coord = df.geometry.values[0].split('(')[1].split(' ')[0]
-        if float(first_coord) > 180:
-            expected_crs = 'EPSG:28992'
-        else:
-            expected_crs = 'EPSG:4326'
-        return (
-            df
-                .assign(geometry=lambda d: d.geometry.apply(wkt.loads))
-                .pipe(gpd.GeoDataFrame, geometry='geometry', crs=expected_crs)
-                .assign(geometry=lambda d: d.geometry.to_crs('EPSG:4326'))
-        )
-
 
 class TrackMap(folium.Map):
     """
     Plotting functionality based on folium maps, designed for plotting objects on the Dutch railways.
-    This object is designed to work as a context manager, where every element can be added to the TrackMap while it
-    is opened.
+    Plottable objects are based on geopandas GeoDataFrames; with the function plottable they can be converted to
+    a PlotObject that can be used for this class.
+    Outputs are automatically zoomed in on what data is requested and include an aerial photograph of the Netherlands.
+    Maps can be shown directly, or saved as .html files.
     """
 
     def __init__(self, objects: List[PlotObject] = [], **kwargs):
         """
-        Set up a TrackMap object
+        Set up a TrackMap object.
 
         :param objects: A list of plottable objects that can be pre-defined outside the context manager of the
         TrackMap
         :return: A TrackMap object
         """
 
         super().__init__(location=[52, 5], zoom_start=8, max_zoom=30, max_native_zoom=30, tiles=None, **kwargs)
-        self._add_luchtfoto()
+        self._add_aerial_photograph()
```

Comment:
Now the aerial photograph is mandatory, maybe make it an option by passing it as an arg with default = True?

Reply:
Fixed!